internal/manifest/os.go

```diff
@@ -480,13 +492,27 @@ func (p *OS) serialize() osbuild.Pipeline {
 		pipeline.AddStage(osbuild.NewPwqualityConfStage(p.PwQuality))
 	}
 
+	// If subscription settings are included there are 3 possible setups:
+	// - Register the system with rhc and enable Insights
+	// - Register with subscription-manager, no Insights or rhc
+	// - Register with subscription-manager and enable Insights, no rhc
 	if p.Subscription != nil {
-		commands := []string{
-			fmt.Sprintf("/usr/sbin/subscription-manager register --org=%s --activationkey=%s --serverurl %s --baseurl %s", p.Subscription.Organization, p.Subscription.ActivationKey, p.Subscription.ServerUrl, p.Subscription.BaseUrl),
-		}
-		if p.Subscription.Insights {
+		var commands []string
+		if p.Subscription.Rhc {
+			// Use rhc for registration instead of subscription manager
+			commands = []string{fmt.Sprintf("/usr/bin/rhc connect -o=%s -a=%s --server %s", p.Subscription.Organization, p.Subscription.ActivationKey, p.Subscription.ServerUrl)}
+
+			// Always enable Insights when using rhc
 			commands = append(commands, "/usr/bin/insights-client --register")
```

Comment:
rhc will automatically register with insights. So this isn't necessary afaik. Can't hurt though.

Reply:
ok. Could have sworn I read that sequence somewhere, but I can't find it again.